### PR TITLE
Add command and UI to create AP metadata.json

### PR DESCRIPTION
**Pull request by mmokko**
Add command and UI element to create an Action Package metadata.json file into the workspace folder.

I did some debugging on why the Windows is so slow to call the action server executable and noticed that there were some data not given to subprocess, that we give when callin RCC. I added those, but didn't notice any significant difference.

### CODE REVIEW

File: sema4ai/src/sema4ai_code/action_server.py [hunk 1]
```diff
@@ -101,7 +102,7 @@ def __init__(self, action_server_location: str):
 
     def _run_action_server_command(
         self,
-        args: list[str],
+        args: List[str],
```
Comment (fabioz) [inline sema4ai/src/sema4ai_code/action_server.py:105]: Why the change here? 

i.e.: Python 3.10 is currently a requirement, so, `list` should usually be preferred over `List`...

File: sema4ai/src/sema4ai_code/action_server.py [hunk 2]
```diff
@@ -515,3 +520,34 @@ def package_set_changelog(
             success=False,
             message=f"Error updating the changelog for package to Control Room.\n{command_result.message or ''}",
         )
+
+    def package_metadata(self, input_dir: str, output_dir) -> ActionResult:
```
Comment (fabioz) [inline sema4ai/src/sema4ai_code/action_server.py:524]: Add type for `output_dir`.

File: sema4ai/vscode-client/src/viewsRobots.ts [hunk 3]
```diff
@@ -284,6 +284,15 @@ export class RobotsTreeDataProvider implements vscode.TreeDataProvider<RobotEntr
...
                         "tooltip": "Create the Action Package .zip file to the workspace folder",
                     },
+                    {
+                        "label": "Create Metadata File",
+                        "uri": element.uri,
+                        "robot": element.robot,
+                        "iconPath": "json",
+                        "type": RobotEntryType.PackageMetadataToWorkspace,
+                        "parent": element,
+                        "tooltip": "Create metadata.json file to the Action Package folder",
```
Comment (fabioz) [inline sema4ai/vscode-client/src/viewsRobots.ts:294]: Suggestion: rename to:

`Create metadata.json file in Action Package folder`

**Review by fabioz — APPROVED** [review]
Just a few minors...

File: sema4ai/src/sema4ai_code/action_server.py [hunk 4]
```diff
@@ -515,3 +520,34 @@ def package_set_changelog(
...
+            "package",
+            "metadata",
+            "--input-dir",
+            input_dir,
+            "--output-file",
+            str(output_file_path),
+        ]
+
+        command_result = self._run_action_server_command(
+            args, timeout=ONE_MINUTE_S * 15
```
Comment (fabioz) [inline sema4ai/src/sema4ai_code/action_server.py:544]: I think I'd actually make the timeout 60 minutes (low end machines with bad net connection can suffer here).

**Comment by fabioz** [comment]
Related to:

> I did some debugging on why the Windows is so slow to call the action server executable and noticed that there were some data not given to subprocess, that we give when callin RCC. I added those, but didn't notice any significant difference.

One thing I've seen happen (and I have no idea why) is that when running in VSCode sometimes a process can get stuck if a Popen is done without closing the `stdin` (or sometimes it'd just be very slow to start).

See: `sema4ai_ls_core.basic.launch_and_return_future` / `sema4ai_ls_core.basic._stdin_write` as a reference.

**Comment by mmokko** [comment]
> launch_and_return_future

Thank you, it seems that I have the same issue. I'll add the same workaround for the action server caller.
